### DISCUSSION TITLE
Elasticsearch Parsing

### DISCUSSION THREAD

**Discussion by gcsrellis** [Security-Onion-Solutions/securityonion#2278] [Unsupported Versions] [answered]
I've sent syslog to my mastersearch node.  I opened up the firewall and it is receiving it.  I can see the logs in kibana under syslog.  According to the docs, the incoming logs are sent to elasticsearch via logstash and elastic parses it via its parsers in /opt/so/conf/elasticsearch/ingest.   I'm assuming that the elasticsearch ingest parser "syslog" is the one trying to parse these logs because when I drop down the table in kibana it says "event.dataset syslog" "event.module syslog".  Am I looking in the right place to modify the parser to work with my logs?

**Comment by weslambert** [comment, marked as answer]
You can create your own pipeline that keys off certain characteristics of your logs after they have been initially processed by the syslog pipeline.  The default files are in `/opt/so/saltstack/default/salt/elasticsearch/files/ingest`.  Any custom files should be copied to `/opt/so/saltstack/local/salt/elasticsearch/files/ingest`.  https://docs.securityonion.net/en/latest/elasticsearch.html#parsing
